src/main/java/app/models/Professor.java

```diff
@@ -72,15 +72,60 @@ public void deleteProject(Project project)
         }
     }
 
+    /**
+     * Delete the project, removing it from the Professor's and the Project Coordinator's
+     * list of projects.
+     * @param id ID of the project to delete
+     */
+    public void deleteProject(Long id)
+    {
+        for (Project project: projects)
+        {
+            if (project.getId().equals(id))
+            {
+                projectCoordinator.removeProjectFromList(project);
+                this.removeProjectFromList(project);
+            }
+        }
+    }
+
 
     /**
-     * Archive the project, but keep it in the Professor's list of projects
-     * @param project The project to archive
+     * Toggle the project, but keep it in the Professor's list of projects
+     * @param project The project to archive/unarchive
      */
-    public void archiveProject(Project project)
+    public void toggleArchive(Project project)
     {
         if (projects.contains(project))
-            project.setIsArchived(true);
+        {
+            if (!project.isArchived())
+                project.setIsArchived(true);
+            else
+                project.setIsArchived(false);
+        }
+
+        else
+            project.setIsArchived(false);
+    }
+
+    /**
+     * Toggle the project based on ID, but keep it in the Professor's list of projects
+     * @param id ID of the project to archive/unarchive
+     */
+    public void toggleArchive(Long id)
+    {
+        for (Project project: projects)
+        {
+            if (project.getId().equals(id))
+            {
+                if (!project.isArchived())
+                    project.setIsArchived(true);
+                else
+                    project.setIsArchived(false);
+
+                System.out.println("Project Archive = " + project.isArchived());
```

Comment:
🔥 the println.

Also call toggleArchive(project) when you get the right project.